test/vms.volumes.test.js

```diff
@@ -0,0 +1,308 @@
+/*
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+/*
+ * Copyright 2017 Joyent, Inc.
+ */
+
+var assert = require('assert-plus');
+var uuid = require('libuuid');
+var vasync = require('vasync');
+
+var common = require('./common');
+var waitForValue = common.waitForValue;
+
+var client;
+
+var ADMIN_USER_UUID = common.config.ufdsAdminUuid;
+var ADMIN_FABRIC_NETWORK;
+var VMAPI_ORIGIN_IMAGE_UUID;
+var SERVER;
+var TEST_VOLUMES_NAME_PREFIX = 'vmapitest-volumes-';
+
+function getVmPayloadTemplate() {
+    return {
+        alias: 'vmapitest-volumes-' + uuid.create().split('-')[0],
+        owner_uuid: ADMIN_USER_UUID,
+        image_uuid: VMAPI_ORIGIN_IMAGE_UUID,
+        server_uuid: SERVER.uuid,
+        networks: [ { uuid: ADMIN_FABRIC_NETWORK.uuid } ],
+        brand: 'joyent-minimal',
+        billing_id: '00000000-0000-0000-0000-000000000000',
+        ram: 64,
+        quota: 10,
+        /*
+         * Not setting a cpu_cap here would break the ability for Triton to
+         * provision any VM with a non-null cpu_cap, since a mix of capped and
+         * cap-less VMs is not allowed by the allocation system.
```

Comment:
Right, but two things:

1. Not setting a cap is _guaranteed_ to have issues, setting a cap has issues, as you pointed out, only in _some_ cases that we don't support when running Triton tests suites.

2. I assume we don't support a setup with both Triton and manta sharing provisionable CNs.